src/routing/route.cc

```diff
@@ -190,31 +200,164 @@ LaneSRangeRelation Route::ComputeLaneSRangeRelation(const api::LaneSRange& lane_
       get_lane_s_range_road_position(lane_s_range_b_index->first, lane_s_range_b_index->second, kEnd);
 
   return kRelativePositionToPreceedingLaneSRange[static_cast<size_t>(
-      ComputeRelativePositionFor(lane_s_range_a_road_pos, lane_s_range_b_road_pos, tolerance))];
+      ComputeRelativePosition(lane_s_range_a_road_pos, lane_s_range_b_road_pos, tolerance))];
 }
 
-std::optional<Route::LaneSRangeIndex> Route::FindLaneSRangeIndex(const api::LaneSRange& lane_s_range) const {
-  auto is_lane_s_range_contained = [tolerance = road_network_->road_geometry()->linear_tolerance()](
-                                       const api::LaneSRange& lane_s_range_a, const api::LaneSRange& lane_s_range_b) {
-    if (lane_s_range_a.lane_id() != lane_s_range_b.lane_id()) {
-      return false;
-    }
-    const bool s1_is_in_range = lane_s_range_a.s_range().s1() + tolerance >= lane_s_range_b.s_range().s1() &&
-                                lane_s_range_a.s_range().s0() - tolerance <= lane_s_range_b.s_range().s1();
-    const bool s0_is_in_range = lane_s_range_a.s_range().s1() + tolerance >= lane_s_range_b.s_range().s0() &&
-                                lane_s_range_a.s_range().s0() - tolerance <= lane_s_range_b.s_range().s0();
-    return s1_is_in_range && s0_is_in_range;
+api::LaneSRoute Route::ComputeLaneSRoute(const api::RoadPosition& start_position) const {
+  MALIPUT_THROW_UNLESS(start_position.lane != nullptr);
+
+  const api::RoadPosition& end_road_position = end_route_position();
+  // Creates a point like api::LaneSRange to find the appropriate LaneSRangeIndex of the start
+  // and end positions.
+  const api::LaneSRange end_lane_s_range_point(end_road_position.lane->id(),
+                                               api::SRange(end_road_position.pos.s(), end_road_position.pos.s()));
+  const api::LaneSRange start_lane_s_range_point(start_position.lane->id(),
+                                                 api::SRange(start_position.pos.s(), start_position.pos.s()));
+
+  const std::optional<LaneSRangeIndex> start_lane_s_range_index = FindLaneSRangeIndex(start_lane_s_range_point);
+  MALIPUT_THROW_UNLESS(start_lane_s_range_index.has_value());
+
+  const std::optional<LaneSRangeIndex> end_lane_s_range_index = FindLaneSRangeIndex(end_lane_s_range_point);
+  MALIPUT_THROW_UNLESS(end_lane_s_range_index.has_value());
+
+  // Inserts @p lane_s_range_index into @p lane_s_range at the beginning.
```

Comment:
"lane_s_range" should be "lane_s_ranges."

Reply:
Done.